doc/md/tutorial-todo-gql-paginate.md

```diff
@@ -0,0 +1,235 @@
+and tagged (using Git) in each step. If you want to skip the basic setup and start with the initial version of the GraphQL
+server, you can clone the repository and checkout `v0.1.0` as follows:
+
+```console
+git clone git@github.com:a8m/ent-graphql-example.git
+cd ent-graphql-example 
+go run ./cmd/todo/
+```
+
+
+## Add Annotations To Schema
+
+Ordering can be defined on any comparable field of ent by annotating it with `entgql.Annotation`.
+Note that the given `OrderField` name must match its enum value in GraphQL schema (see
+[next section](#define-ordering-types-in-graphql-schema) below).
+
+```go
+func (Todo) Fields() []ent.Field {
+    return []ent.Field{
+		field.Text("text").
+			NotEmpty().
+			Annotations(
+				entgql.OrderField("TEXT"),
+			),
+		field.Time("created_at").
+			Default(time.Now).
+			Immutable().
+			Annotations(
+				entgql.OrderField("CREATED_AT"),
+			),
+		field.Enum("status").
+			NamedValues(
+				"InProgress", "IN_PROGRESS",
+				"Completed", "COMPLETED",
+			).
+			Default("IN_PROGRESS").
+			Annotations(
+				entgql.OrderField("STATUS"),
+			),
+		field.Int("priority").
+			Default(0).
+			Annotations(
+				entgql.OrderField("PRIORITY"),
+			),
+    }
+}
+```
+
+## Define Types In GraphQL Schema
+
+Next, we need to define the ordering types along with the [Relay Connection Types](https://relay.dev/graphql/connections.htm#sec-Connection-Types)
+in the GraphQL schema:
+
+```graphql
+# Define a Relay Cursor type:
+# https://relay.dev/graphql/connections.htm#sec-Cursor
+scalar Cursor
+
+type PageInfo {
+    hasNextPage: Boolean!
+    hasPreviousPage: Boolean!
+    startCursor: Cursor
+    endCursor: Cursor
+}
+
+type TodoConnection {
+    totalCount: Int!
+    pageInfo: PageInfo!
+    edges: [TodoEdge]
+}
+
+type TodoEdge {
+    node: Todo
+    cursor: Cursor!
+}
+
+# These enums are matched the entgql annotations in the ent/schema.
+enum TodoOrderField {
+    CREATED_AT
+    PRIORITY
+    STATUS
+    TEXT
+}
+
+enum OrderDirection {
+    ASC
+    DESC
+}
+
+input TodoOrder {
+    direction: OrderDirection!
+    field: TodoOrderField
+}
+```
+
+Note that the naming must take the form of `<T>OrderField` / `<T>Order` for `autobind`ing to the generated ent types.
+Alternatively [@goModel](https://gqlgen.com/config/#inline-config-with-directives) directive can be used for manual type binding.
+
+## Add Pagination Support For Query
+
+```graphql
+type Query {
+    todos(
+        after: Cursor
+        first: Int
+        before: Cursor
+        last: Int
+        orderBy: TodoOrder
+    ): TodoConnection
+}
+```
+That's all for the GraphQL schema changes, let's run `gqlgen` code generation.
+
+## Update The GraphQL Resolver
+
+After changing our Ent and GraphQL schemas, we're ready to run the codegen and use the `Paginate` API:
+
+```console
+go generate ./...
+```
+
+If you got an error, ignore it and head over to the `Todos` resolver and update it to pass `orderBy` argument
```

Comment:
if there's an expected error here it would be useful to tell what it is so it ends up in search engine results and more transparent to the reader

Reply:
👍